concepts/eventing-code-guidelines.md

```diff
@@ -0,0 +1,1155 @@
+logrus does not enforce using structured logging but you can use it. 
+Citing from the logrus github page it is clear that zap is the modern alternative to logrus:
+> Logrus is in maintenance-mode. We will not be introducing new features. It’s simply too hard to do in a way that won’t break many people’s projects, which is the last thing you want from your Logging library (again...).
+
+> Many fantastic alternatives have sprung up. Logrus would look like those, had it been re-designed with what we know about structured logging in Go today. Check out, for example, Zerolog, Zap, and Apex.
+
+For these reasons `uber-go/zap` is the `preferred` structured logging library.
+
+Furthermore consider using `github.com/kyma-project/kyma/common/logging/logger` which provides further `abstraction` over `uber-go/zap`.
+
+<!-- voting options:
+1. [ueber-go/zap](https://github.com/uber-go/zap) // proposed by guidelines
+2. [logrus](https://github.com/sirupsen/logrus) 
+-->
+
+#### See Also
+
+- [Kyma Logging Proposal](https://github.com/kyma-project/community/blob/main/concepts/observability-consistent-logging/improvement-of-log-messages-usability.md)
+
+## Documentation Guidelines
+
+Code is read many times but sometimes only written once. Therefore you should always make sure that you follow these guidelines.
+
+**Guidelines**:
+1. Add `documentation` to `all exported` functions/variables/types etc. Comments on private elements are also welcome because they help the reader.
+1. Assume that there is a generated html `version` of our `docs` on <https://pkg.go.dev>, e.g. [here](https://pkg.go.dev/github.com/kyma-project/kyma/components/eventing-controller@v0.0.0-20220204130403-a0b8b10f369d#section-readme) for the eventing-controller.
+1. Comments on exported functions/variables/types shall `start` with the name of the element and `end` with a dot (`.`). This forces you to write real sentences.
+
+**Example**:
+
+```go
+// source: https://go.dev/blog/godoc
+
+// Fprint formats using the default formats for its operands and writes to w.
+// Spaces are added between operands when neither is a string.
+// It returns the number of bytes written and any write error encountered.
+func Fprint(w io.Writer, a ...interface{}) (n int, err error) {
+```
+
+If you are in doubt how to write the godoc properly, run `godoc` and check the genrated documentation:
+```bash
+$ go install golang.org/x/tools/cmd/godoc@latest
+$ godoc -http=:8080
+```
+
+### Bad Practice
+
+**Example1**: Using backticks when referring to arguments in the method siganture
+
+<details>
+	<summary>Don't</summary>
+
+In the example `w` is the first argument to the function `Fprint`.
+It is not necessary to put p in backticks. Goland supports jumping to the element defintion (at least sometimes :-D). Using backticks however breaks the feature.
```

Comment:
you mean "put w in backticks"?

Reply:
good spot, didn't notice that :D

But let's put the p in backticks: 
bapptipps! packpicks!

Reply:
@raypinto Jep, will fix it. Thank you 😃